src/pixelator/cli/logging.py

```diff
@@ -0,0 +1,276 @@
+import time
+import typing
+import warnings
+from pathlib import Path
+from typing import Callable
+
+import click
+from numba import NumbaDeprecationWarning
+
+from pixelator.types import PathType
+
+root_logger = logging.getLogger()
+pixelator_root_logger = logging.getLogger("pixelator")
+
+
+# Silence deprecation warnings
+warnings.simplefilter("ignore", category=NumbaDeprecationWarning)
+warnings.filterwarnings(
+    "ignore",
+    message="geopandas not available. Some functionality will be disabled.",
+    category=UserWarning,
+)
+
+
+# Disable matplot lib debug logs (that will clog all debug logs)
+logging.getLogger("matplotlib.font_manager").setLevel(logging.ERROR)
+logging.getLogger("matplotlib.ticker").setLevel(logging.ERROR)
+logging.getLogger("numba").setLevel(logging.ERROR)
+
+
+# ------------------------------------------------------------
+# Click logging
+# ------------------------------------------------------------
+
+LOGGER_KEY = __name__ + ".logger"
+DEFAULT_LEVEL = logging.INFO
+
+
+class StyleDict(typing.TypedDict):
+    """Style dictionary for kwargs to `click.style`."""
+
+    fg: str
+
+
+class ColorFormatter(logging.Formatter):
+    """Click formatter with colored levels"""
+
+    colors: dict[str, StyleDict] = {
+        "debug": StyleDict(fg="blue"),
+        "info": StyleDict(fg="green"),
+        "warning": StyleDict(fg="yellow"),
+        "error": StyleDict(fg="red"),
+        "exception": StyleDict(fg="red"),
+        "critical": StyleDict(fg="red"),
+    }
+
+    def format(self, record: logging.LogRecord) -> str:
+        """Format a record with colored level.
+
+        :param record: The record to format.
+        :returns str: A formatted log record.
+        """
+        if not record.exc_info:
+            level = record.levelname.lower()
+            msg = record.getMessage()
+            if level in self.colors:
+                timestamp = self.formatTime(record, self.datefmt)
+                colored_level = click.style(
+                    f"{level.upper():<10}", **self.colors[level]
+                )
+                prefix = f"{timestamp} [{colored_level}]  "
+                msg = "\n".join(prefix + x for x in msg.splitlines())
+            return msg
+        return logging.Formatter.format(self, record)
+
+
+class DefaultCliFormatter(logging.Formatter):
+    """Click formatter with colored levels"""
+
+    def format(self, record: logging.LogRecord) -> str:
+        """Format a record for CLI output."""
+        if not record.exc_info:
+            level = record.levelname.lower()
+            msg = record.getMessage()
+
+            if level == "info":
+                return msg
+
+            return f"{level.upper()}: {msg}"
+        return logging.Formatter.format(self, record)
+
+
+class ClickHandler(logging.Handler):
+    """Click logging handler.
+
+    Messages are forwarded to stdout using `click.echo`.
+    """
+
+    def __init__(self, *args, use_stderr: bool = True, **kwargs):
```

Comment:
I have refactored this to just explicitly use the level arg instead of args and kwargs since that is the only thing that gets passed to the base class.